source/client/Controller.py

```diff
@@ -15,42 +15,49 @@ class Controller(ConnectionListener):
 
     def __init__(self, clientState):
         self._state = clientState
-        self.prepared_cards = {} #This is the dict of cards prepared to be played
+        self.prepared_cards = {}     #This is the dict of cards prepared to be played
         self.setName()
         self.ready = False
         self.note = "Game is beginning."
+        # variables needed for games with Shared_Board == True (i.e. Liverpool):
+        self.Meld_Threshold = self._state.rules.Meld_Threshold
+        self.player_index = 0
+        self.visible_scards = [{}] # if Shared_Board both HandView and TableView use data:visible cards.
 
     ### Player Actions ###
     def setName(self):
         """Set up a display name and send it to the server"""
 
         # to prevent duplicate names, displayname = 'guest' is forbidden.
+        # Forbidden names are defined at the beginning of this controller.
         # May as well allow other names to be forbidden, too (for fun :) )
         # if name is in list of forbidden names, then changeName is called.
         displayName = input("Enter a display name: ")
         if displayName in Forbidden_Names:
-            self.note = "Sorry, but that name is forbidden."
-            self.changeName()
+            self.note = "Sorry, but that name ('"+displayName+"') is forbidden."
```

Comment:
NITPICK I would say "the name <displayName>" and not "that name <displayName>", or maybe "that name, <displayName>," with commas. It looks wrong to me the way it is (TOTAL nitpick, I know)